src/layman/layer/filesystem/input_style.py

```diff
@@ -85,18 +83,16 @@ def get_metadata_comparison(username, publication_name):
     pass
 
 
-def get_style_type_from_xml_file(style_path):
-    if style_path and os.path.exists(style_path):
-        xml_tree = etree.parse(style_path)
-        root_tag = xml_tree.getroot().tag
+def get_style_type_from_file_storage(file_storage):
+    if file_storage:
+        xml = file_storage.read()
+        file_storage.seek(0)
+        xml_tree = etree.fromstring(xml)
+        root_tag = xml_tree.tag
         root_attribute = etree.QName(root_tag).localname
         result = next((sd for sd in layer.STYLE_TYPES_DEF if sd.root_element == root_attribute), None)
         if not result:
             raise LaymanError(46)
     else:
```

Comment:
Probably redundant

Reply:
It is not and I like, that the logic 'no style file means SLD style' is at one place.